nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -182,8 +186,18 @@
 
     protected boolean isLatestVersionDisabled = false;
 
+    protected NuxeoPrincipal principal;
```

Comment:
Storing the principal in the `DBSSession` (and the `SQLSession`) is really not going to work with further planned refactoring where the session *must* be independent of the context because it can be cached.
So it should be simpler to just pass the principal down to the `remove()` method that needs it.
And even then it may not really be needed, `remove()` is done at a level where the security (and integrity/constraints checks) have already beed done so subsequent bulk removes can be done as a system user.